.github/workflows/conformance-aks.yaml

```diff
@@ -355,16 +355,41 @@ jobs:
     runs-on: ubuntu-latest
     needs: installation-and-connectivity
     steps:
+      # TODO remove list and filter artifact steps if this PR https://github.com/actions/upload-artifact/pull/521 is merged
+      - name: List All Artifacts
+        id: list_artifacts
+        run: |
+          echo "Fetching artifacts"
+          curl --fail --show-error --silent \
+               -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+               -H "Accept: application/vnd.github.v3+json" \
+               "https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/artifacts" \
+               | jq -r '.artifacts[] | .name' > artifacts_list.json
+      - name: Filter Artifacts
+        id: filter_artifacts
+        run: |
+          echo "Filtered Artifacts starts with cilium-sysdumps:"
+          if grep -E "^cilium-sysdumps-" artifacts_list.json; then
+            echo "sysdumps=true" >> $GITHUB_OUTPUT
+          else
+            echo "::warning::No matching cilium-sysdumps-* artifacts found."
+          fi
+          echo "Filtered Artifacts starts with cilium-junits:"
+          if grep -E "^cilium-junits-" artifacts_list.json; then
+            echo "junits=true" >> $GITHUB_OUTPUT
+          else
+            echo "::warning::No matching cilium-junits-* artifacts found."
+          fi
```

Comment:
Should we offload this to a reusable action instead, to centralize this code piece?

Reply:
OK saw the discussion above, yes I agree with @giorio94 that this should be made into a reusable workflow, especially so if upstream is not intending on merging the required PR.

Reply:
I will convert this to a reusable workflow with a follow-up PR.